datastore/storagebackend/postgresql/putobservations.go

```diff
@@ -370,6 +373,10 @@ func (sbe *PostgreSQL) PutObservations(request *datastore.PutObsRequest) error {
 			return fmt.Errorf("getObsTime() failed: %v", err)
 		}
 
+		if obsTime.AsTime().Before(loTime) || obsTime.AsTime().After(hiTime) {
```

Comment:
Due to rounding of hiTime down to seconds, this is very likely to reject observations?

Also, I think this should throw an error, especially for observations in the future. Just dropping the observation is unexpected behaviour from the user point of view.

Reply:
Same argument as above: I think this leeway should be configured with `HITIME` env.var. For example, setting it to `-3600` would allow the ingestor to insert observations as much as one hour into the future. Which seems weird but doesn't really harm anyone?

Reply:
Do you agree this should grow an error.

Reply:
I currently have the branch _putobs_summary_ in progress that will return the number of _accepted_ and _rejected_ observations. Thus the client will informed of such events by inspecting the response of a successful call (if the call fails it would be for more serious reasons). The risk is of course that a bad-behaving/careless client will more easily be able to send a lot of out-of-range observations, thus effectively wasting resources. So maybe better to have the call as a whole fail? Without even inserting/updating the observations that were within the valid time range? What do you think?

Reply:
Hmm, good question. I am leaning towards failing the whole thing without inserting anything if any of the observations are not within the time range. It will be easier for the client to work with. Even if you get the number of accepted and rejected observations, it is still not clear how to proceed, as you don't know which ones failed.

Reply:
Very good point. The ingestor would actually have to iterate until nrejected drops to zero. And in each iteration wasting resources by updating the observations that weren't rejected with the same value over again! So I'll implement an all-or-nothing behavior here.